src/test/kotlin/eu/europa/ec/eudi/sdjwt/vc/SdJwtVcVerifierTest.kt

```diff
@@ -106,21 +107,14 @@ class SdJwtVcVerifierTest {
 
     @Test
     fun `keySource should return a Metadata when iss is a https url`() {
-        val expectedSource = SdJwtVcIssuerPublicKeySource.Metadata(Url("https://example.com"), null)
-        testForMetaDataSource(expectedSource)
-    }
-
-    @Test
-    fun `keySource should return a Metadata when iss is a https url and kid is provided`() {
-        val expectedSource = SdJwtVcIssuerPublicKeySource.Metadata(Url("https://example.com"), "some-kid")
+        val expectedSource = SdJwtVcIssuerPublicKeySource.Metadata(Url("https://example.com"))
         testForMetaDataSource(expectedSource)
     }
 
     private fun testForMetaDataSource(expectedSource: SdJwtVcIssuerPublicKeySource.Metadata) {
         val jwt = run {
             val header = JWSHeader.Builder(JWSAlgorithm.ES256).apply {
                 type(JOSEObjectType(SD_JWT_VC_TYPE))
-                expectedSource.kid?.let { keyID(it) }
             }.build()
```

Comment:
Why was kid removed?

Reply:
`kid` was populated using the value from `Metadata`. Since `Metadata` no longer contains `kid` it was removed. We could add it as an extra parameter to the method, if you believe it's required.

Reply:
Indeed there is no reference to `kid` in the resolution section.

There is though the following [recommendation](https://www.ietf.org/archive/id/draft-ietf-oauth-sd-jwt-vc-06.html#name-jwt-vc-issuer-metadata-resp):

>It is RECOMMENDED that the JWT contains a kid JWT header parameter that can be used to look up the public key in the JWK Set included by value or referenced in the JWT VC Issuer Metadata

I would say, to keep it, although in our current implementation we delegate this responsibility to nimbus.
An alternative, would be to peek from the JWK set returned via issuer's metadata, the corresponding JWK using the kid.

Please keep the `kid` and leave the alternative for another PR.